docs/upgrading/android-csproj.md

```diff
@@ -0,0 +1,350 @@
+
+### Preview testing
+
+For the latest instructions on preview testing and sample projects,
+see the [net6-samples][net6-samples] repo.
+
+[net6-samples]: https://github.com/xamarin/net6-samples
+
+## Package Versioning Scheme
+
+This is the package version scheme: `OS-Major.OS-Minor.InternalRelease[-prereleaseX]+sha.1b2c3d4`.
+
+* Major: The highest stable API level, such as 30. On Apple platforms, this is the major OS version.
+* Minor: Always 0 for Android. On Apple platforms, this is the minor OS version.
+* Patch: Our internal release version based on `100` as a starting point.
+    * Service releases will bump the last two digits of the patch version
+    * Feature releases will round the patch version up to the nearest 100
+      (this is the same as bumping the first digit of the patch version, and
+      resetting the last two digits to 0).
+    * This follows [how the dotnet SDK does it][1].
+* Pre-release: Optional (e.g.: Android previews, CI, etc.)
+    * For CI we use a `ci` prefix + the branch name (cleaned up to only be
+      alphanumeric) + the commit distance (number of commits since any of the
+      major.minor.patch versions changed).
+        * Example: `30.0.100-ci.master.1234`
+        * Alphanumeric means `a-zA-Z0-9-`: any character not in this range
+          will be replaced with a `-`.
+    * Pull requests have `pr` prefix, followed by `gh`+ PR number + commit
+      distance.
+        * Example: `30.0.200-ci.pr.gh3333.1234`
+    * If we have a particular feature we want people to subscribe to (such as
+      an Android preview release), we publish previews with a custom pre-release
+      identifier:
+        * Example: `30.0.100-android-r.beta.1`
+        * This way people can sign up for only official previews, by
+          referencing `*-android-r.beta.*`
+        * It's still possible to sign up for all `android-r` builds, by
+          referencing `*-ci.android-r.*`
+* Build metadata: Required Hash
+    * This is `sha.` + the short commit hash.
+        * Use the short hash because the long hash is quite long and
+          cumbersome. This leaves the complete version open for duplication,
+          but this is extremely unlikely.
+    * Example: `30.0.100+sha.1a2b3c`
+    * Example (CI build): `30.0.100-ci.master.1234+sha.1a2b3c`
+    * Since the build metadata is required for all builds, we're able to
+      recognize incomplete version numbers and determine if a particular
+      version string refers to a stable version or not.
+        * Example: `30.0.100`: incomplete version
+        * Example: `30.0.100+sha.1a2b3c`: stable
+        * Example: `30.0.100-ci.d17-0.1234+sha.1a2b3c`: CI build
+        * Example: `30.0.100-android-r.beta.1+sha.1a2b3c`: official
+          preview
+            * Technically it's possible to remove the prerelease part, but
+              we’d still be able to figure out it’s not a stable version by
+              using the commit hash.
```

Comment:
This whole section is about versioning of our packages. It might not be that relevant, maybe could delete it all?